cfme/intelligence/reports/menus.py

```diff
@@ -180,6 +194,41 @@ def manage_folder(self, group, folder=None):
             view.manager.commit()
             view.save_button.click()
 
+    def move_reports(self, group, folder, subfolder, *reports):
+        """ Moves a list of reports to a given menu
+        Args:
+            group: User group
+            folder: Parent of the subfolder where reports are to be moved.
+            subfolder: Subfolder under which the reports are to be moved.
+            reports: List of reports that are to be moved.
+        """
+        reports = list(reports)
+        cancel_view = ""
+
+        with self.manage_subfolder(group, folder, subfolder) as selected_menu:
+            selected_options = selected_menu.parent_view.report_select.all_options
+
+            if set(selected_options) & set(reports):
+                # If the report is already present, we stay on the same page after fill is complete.
+                cancel_view = self.create_view(EditReportMenusView)
+
+            # fill method replaces all the options in all_options with the value passed as argument
+            # We do not want to replace any value, we just want to move the new reports to a given
+            # menu. This is a work-around for that purpose.
+            reports.extend(selected_options)
+            selected_menu.parent_view.report_select.fill(reports)
+
+        if cancel_view:
+            assert cancel_view.is_displayed
+            cancel_view.cancel_button.click()
+            message = 'Edit of Report Menu for role "{}" was cancelled by the user'
+        else:
+            message = 'Report Menu for role "{}" was saved'
```

Comment:
Do we not have to click any save buttons here?

Reply:
Don't need to click any buttons, `manage_subfolder` context manager takes care of that.